Directory.Build.props

```diff
@@ -1,28 +1,41 @@
 <Project>
+  <!-- this allows us to depend on explicit package versions or version ranges while also using local project references -->
+  <!-- justification being versioning different packages differently depending on what changes are made, in line with our versioning rules -->
+  <Target Name="UseExplicitPackageVersions" BeforeTargets="GenerateNuspec">
+    <ItemGroup>
+      <_ProjectReferenceWithExplicitPackageVersion Include="@(ProjectReference->'%(FullPath)')" Condition="'%(ProjectReference.PackageVersion)' != ''" />
+      <_ProjectReferenceWithExactPackageVersion    Include="@(ProjectReference->'%(FullPath)')" Condition="'%(ProjectReference.ExactVersion)' == 'true'" />
+      <_ProjectReferenceWithReassignedVersion      Include="@(_ProjectReferencesWithVersions)"  Condition="'%(Identity)' != '' And '@(_ProjectReferencesWithVersions)' == '@(_ProjectReferenceWithExplicitPackageVersion)'" ProjectVersion="@(_ProjectReferenceWithExplicitPackageVersion->'%(PackageVersion)')" />
+      <_ProjectReferenceWithReassignedVersion      Include="@(_ProjectReferencesWithVersions)"  Condition="'%(Identity)' != '' And '@(_ProjectReferencesWithVersions)' == '@(_ProjectReferenceWithExactPackageVersion)'"    ProjectVersion="@(_ProjectReferencesWithVersions->'%(ProjectVersion)')" />
+      <_ProjectReferencesWithVersions              Include="@(_ProjectReferenceWithReassignedVersion)" Remove="@(_ProjectReferenceWithReassignedVersion)" />
+    </ItemGroup>
+  </Target>
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
     <Nullable>enable</Nullable>
     <LangVersion>11</LangVersion>
-    <Authors>Naamloos, afroraydude, DrCreo, Death, TiaqoY0, Axiom, Emzi0767, IDoEverything, Velvet, OoLunar, akiraveliara, DSharpPlus contributors</Authors>
-    <Company>DSharpPlus Contributors</Company>
-    <PackageProjectUrl>https://github.com/DSharpPlus/DSharpPlus</PackageProjectUrl>
-    <RepositoryUrl>https://github.com/DSharpPlus/DSharpPlus</RepositoryUrl>
-    <RepositoryType>Git</RepositoryType>
-    <PackageLicenseExpression>MPL</PackageLicenseExpression>
-    <PackageIcon>dsharpplus.png</PackageIcon>
-    <PackageIconUrl>https://raw.githubusercontent.com/DSharpPlus/DSharpPlus/master/logo/dsharpplus.png</PackageIconUrl>
     <GenerateDocumentationFile>True</GenerateDocumentationFile>
-    <NoWarn>CS1591</NoWarn>
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
-
     <!-- build and versioning setup -->
     <ManagePackageVersionsCentrally>True</ManagePackageVersionsCentrally>
     <UseArtifactsOutput>true</UseArtifactsOutput>
-
-    <!-- SourceLink -->
+  </PropertyGroup>
+  <!-- sourceLink -->
+  <PropertyGroup Condition="$(Configuration) == 'Debug'">
```

Comment:
sourcelink is expressly a release concept

Reply:
![изображение](https://github.com/DSharpPlus/DSharpPlus/assets/60395110/666d34f3-9022-4734-b6fe-8d34e4ec0ad7)
docfx uses sourcelink for documentation when you use .dll files for docs generation, so it's... kinda useful? (not really, because we use .sln)

Reply:
i'd rather not use docfx at all if we can help it, fwiw. but source-link is specifically designed to allow the end user of a release build to debug their interaction with library code, disabling it in release defies the entire point of the feature.

Reply:
> not use docfx
should i remind you about how all alternatives are even worse, sadly

Reply:
I misunderstood what sourcelink did. I'll go ahead and move it to the top property group then